### PR TITLE
feat: checkbox common component

### PR DESCRIPTION
**Pull request by Tanyashinke**

### CODE REVIEW

File: frontend/src/app/libs/common-components-demo/components/checkbox-demo/checkbox-demo.component.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+					[options]="checkboxGroupData.options" 
+					[theme]=darkTheme
+					[labelField]="checkboxGroupData.labelField" 
+					[valueField]="checkboxGroupData.valueField"
+					(change)="onChange($event)">
+				</app-checkbox-group>
+			</app-demo-state>
+		</div>
+	</app-demo-group>
+
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components-demo/components/checkbox-demo/checkbox-demo.component.html:25]: add new line at the end of file

File: frontend/src/app/libs/common-components-demo/components/checkbox-demo/checkbox-demo.component.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.checkbox-group-demo {
+	&__container {
+		display: block;
+		max-width: 400px;
+	}
+}
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components-demo/components/checkbox-demo/checkbox-demo.component.scss:6]: add new line at the end of file

File: frontend/src/app/libs/common-components/components/checkbox/checkbox.theme.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.mat-checkbox-dark {
+	color: $color-white;
+	@include mat-checkbox-theme ($app-theme-dark);
+
+	&.mat-checkbox-disabled {
+		.mat-checkbox-label {
+			color: $color-white-50;
+		}
+	}
+}
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/checkbox/checkbox.theme.scss:10]: add new line at the end of file

File: frontend/src/app/libs/common-components/components/checkbox/checkbox.component.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+		[value]="isDefined(valueField) ? option[valueField] : (index + 1)"
+		*ngFor="let option of options; let index = index"
+		(change)="check($event)">
+		<span class="checkbox-group__checkbox-label" 
+				[class.checkbox-group__checkbox-label--disabled]="option.disabled" 
+				[class.checkbox-group__checkbox-label--dark-theme]="isDarkTheme">
+			{{option[labelField]}}
+		</span>
+	</mat-checkbox>
+</div>
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/checkbox/checkbox.component.html:17]: add new line at the end of file

File: frontend/src/app/libs/common-components/components/checkbox/checkbox.component.html [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<div class="checkbox-group">
+	<mat-checkbox
+		color="primary"
+		[disabled]="option.disabled"
+		[checked]="option.checked"
+		[class.mat-checkbox-disabled]="option.disabled"
+		[class.mat-checkbox-dark]="isDarkTheme"
+		[value]="isDefined(valueField) ? option[valueField] : (index + 1)"
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/checkbox/checkbox.component.html:8]: move to getter please

**Review by clain-atomi — CHANGES_REQUESTED** [review]

File: frontend/src/app/libs/common-components/components/checkbox/checkbox.component.html (hunk 5 above)
Comment (Tanyashinke) [reply frontend/src/app/libs/common-components/components/checkbox/checkbox.component.html]: I have moved this to the getValue

**Review by clain-atomi — APPROVED** [review]

**Pull request merged**